core/src/main/java/com/sequenceiq/cloudbreak/conf/SecurityConfig.java

```diff
@@ -74,29 +81,55 @@ protected MethodSecurityExpressionHandler createExpressionHandler() {
         private static final String AUTOSCALE_API = API_ROOT_CONTEXT + "/autoscale/**";
 
         @Inject
-        private ResourceServerTokenServices resourceServerTokenServices;
+        private GrpcUmsClient umsClient;
+
+        @Bean
+        public RequestHeaderAuthenticationFilter headerAuthenticationFilter() throws Exception {
+            RequestHeaderAuthenticationFilter requestHeaderAuthenticationFilter = new RequestHeaderAuthenticationFilter();
+            requestHeaderAuthenticationFilter.setPrincipalRequestHeader("x-cdp-actor-crn");
+            requestHeaderAuthenticationFilter.setAuthenticationManager(authenticationManager());
+            requestHeaderAuthenticationFilter.setExceptionIfHeaderMissing(false);
+            requestHeaderAuthenticationFilter.setContinueFilterChainOnUnsuccessfulAuthentication(true);
+            return requestHeaderAuthenticationFilter;
+        }
 
-        @Inject
-        private ScimAccountGroupReaderFilter scimAccountGroupReaderFilter;
+        @Bean
+        public UserDetailsService userDetailsService() {
+            return new CrnUserDetailsService(umsClient);
+        }
 
+        @Bean
         @Override
-        public void configure(ResourceServerSecurityConfigurer resources) {
-            resources.resourceId("cloudbreak");
-            resources.tokenServices(resourceServerTokenServices);
-            resources.tokenExtractor(new CrnTokenExtractor());
+        protected AuthenticationManager authenticationManager() throws Exception {
+            List<AuthenticationProvider> providers = new ArrayList<>(1);
+            providers.add(preauthAuthProvider());
+            return new ProviderManager(providers);
+        }
+
+        @Bean(name = "preAuthProvider")
```

Comment:
Bean name can be specified by the name of the method. If you prefer to use this annotation, then please rename the method to have the same name as the bean, just for making it cleaner.

Reply:
Why is the explicit naming necessary by the way?

Reply:
I'm not sure about it, I agree, I was just not sure if it has any meaning.